tests/ui/issues/issue-12133-3.rs

```diff
@@ -1,4 +1,4 @@
-// run-pass
```

Comment:
This tests dynamic linking. The original issue #12133 also looks like it encountered issue at runtime.

I'm concerned that these changes are applied too broadly and some other subtle things that happen before/after main are being missed.